integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java

```diff
@@ -0,0 +1,177 @@
+ *software distributed under the License is distributed on an
+ *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *KIND, either express or implied.  See the License for the
+ *specific language governing permissions and limitations
+ *under the License.
+ */
+package org.wso2.carbon.esb.rest.test.api;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.axis2.AxisFault;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.clients.template.SequenceTemplateAdminServiceClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+/**
+ * Test templateAdminService related operations
+ */
+public class TemplateAdminService extends ESBIntegrationTest {
+    private SequenceTemplateAdminServiceClient templateAdmin;
+    private static final String templateName = "TemplateAdminService";
+    private  static final String dynamicName = "TemplateUserService";
+    private final String KEY = "conf:/template/dynamicAdminTemplate";
+    private static final String template1 =
+            "<template  xmlns=\"http://ws.apache.org/ns/synapse\" name=\"TemplateAdminService\">\n" + "<parameter name=\"message\"/>\n" + " <sequence>\n" + "<log level=\"custom\">\n"
+                    + "<property name=\"TMPL_MSG \" value= \"in Template Service..\" />\n" + "</log>\n" + "   </sequence>\n" + "</template>";
+    private static final String template2 =
+            "<template  xmlns=\"http://ws.apache.org/ns/synapse\" name=\"TemplateAdminService\">\n" + "<parameter name=\"message\"/>\n" + " <sequence>\n" + "<log level=\"custom\">\n"
+                    + "<property name=\"TMPL_MSG \" value= \"in Template Service Updated..\" />\n" + "</log>\n" + "   </sequence>\n" + "</template>";
+    private static final String template3 =
+            "<template  xmlns=\"http://ws.apache.org/ns/synapse\" name=\"TemplateUserService\">\n" + "<parameter name=\"message\"/>\n" + " <sequence>\n" + "<log level=\"custom\">\n"
+                    + "<property name=\"TMPL_MSG \" value= \"in Template Service Updated..\" />\n" + "</log>\n" + "   </sequence>\n" + "</template>";
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        templateAdmin = new SequenceTemplateAdminServiceClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test template creation service",
+          priority = 1)
+    public void testCreateTemplate() throws Exception {
+        int prevCount = templateAdmin.getTemplatesCount();
+        templateAdmin.addSequenceTemplate(AXIOMUtil.stringToOM(template1));
+        int latestCount = templateAdmin.getTemplatesCount();
+        org.testng.Assert.assertTrue(prevCount < latestCount, "New template not added");
```

Comment:
Do we need "org.testng.Assert.assertTrue" here, as we have already imported the class?

Reply:
This was added through auto import. Will fix it (y)

Reply:
Fixed 5e7fb91734735c829b3ea3c5a2df7acf91b2e166